test/doctool/test-doctool-html.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+const html = require('../../tools/doc/html.js');
+
+// Test data is a list of objects with two properties.
+// The file property is the file path.
+// The html property is some html which will be generated by the doctool.
+// This html will be stripped of all whitespace because we don't currently
+// have an html parser.
+const testData = [
+  {
+    'file': common.fixturesDir + '/sample_document.markdown',
+    'html': '<ol><li>fish</li><li><p>fish</p></li><li><p>Redfish</p></li>' +
+      '<li>Bluefish</li></ol>'
+  },
+  {
+    'file': common.fixturesDir + '/order_of_end_tags_5873.markdown',
+    'html': '<h3>ClassMethod: Buffer.from(array) <span> ' +
+      '<a class="mark" href="#foo_class_method_buffer_from_array" ' +
+      'id="foo_class_method_buffer_from_array">#</a> </span> </h3><div' +
+      'class="signature"><ul><li><code>array</code><a ' +
+      'href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/' +
+      'Reference/Global_Objects/Array" class="type">&lt;Array&gt;</a></li>' +
+      '</ul></div>'
+  },
+];
+
+testData.forEach(function(item) {
+  // Normalize expected data by stripping whitespace
+  const expected = item.html.replace(/\s/g, '');
+
+  fs.readFile(item.file, 'utf8', common.mustCall(function(err, input) {
+    assert.ifErr(err);
```

Comment:
This must be [`assert.ifError`](https://nodejs.org/api/assert.html#assert_assert_iferror_value)